app/tests/workstations_tests/test_session_control.py

```diff
@@ -0,0 +1,79 @@
+import json
+from urllib.parse import urlparse
+
+import pytest
+from django.conf import settings
+from django.views.generic import TemplateView
+
+
+class SessionControlView(TemplateView):
+    template_name = "session_control.html"
+
+    def get_context_data(self, *args, **kwargs):
+        context = super().get_context_data()
+        context.update(
+            {
+                "netloc": settings.DJANGO_LIVE_TEST_SERVER_ADDRESS,
+                "domain": settings.DJANGO_LIVE_TEST_SERVER_ADDRESS.split(":")[
+                    0
+                ],
+            }
+        )
+        return context
+
+
+class WorkstationView(TemplateView):
+    template_name = "workstation.html"
+
+
+class SessionCreationView(TemplateView):
+    template_name = "new_session.html"
+
+
+@pytest.mark.playwright
+def test_viewer_session_control(live_server, page, settings):
+    url = urlparse(live_server.url)
+    settings.DJANGO_LIVE_TEST_SERVER_ADDRESS = f"{url.netloc}"
+    session_create_view = f"{url.scheme}://{url.netloc}/new-session/"
+    session_control_view = f"{url.scheme}://{url.netloc}/session-control/"
+    mock_workstation_view = f"{url.scheme}://{url.netloc}/workstation/"
+    page.goto(session_control_view)
+
+    # Test if a fresh click opens up a new page
+    with page.expect_popup() as viewer_page_info:
+        page.get_by_role("button", name="Launch new Session").click()
+        viewer_page = viewer_page_info.value
+    viewer_page.get_by_text(SessionCreationView.template_name).wait_for()
+
+    # Setup mock acknowledge
+    viewer_page.goto(mock_workstation_view)
+    viewer_page.evaluate("enableMockAcks()")
+
+    # Test if pressing launch indeed sends a session control message
+    page.get_by_role("button", name="Launch new Session").click()
+    received_msg = json.loads(
+        viewer_page.locator("#messages :nth-child(1)").inner_text()
+    )
+    assert "id" in received_msg["sessionControl"]["header"]
+
+    # check that ack message is returned
+    sent_msg = json.loads(
+        viewer_page.locator("#acks :nth-child(1)").inner_text()
+    )
+    assert "acknowledge" in sent_msg["sessionControl"]["header"]
+
+    # Test that if acknowledge sent too late, a new session will be created
+    viewer_page.evaluate("enableMockAcksWithDelay()")
+    page.get_by_role("button", name="Launch new Session").click()
+    viewer_page.get_by_text(SessionCreationView.template_name).wait_for()
+    assert viewer_page.url.startswith(session_create_view)
+
+    # Test if disabling the acknowledgment will result in a new session being created
```

Comment:
I think we should:
* add a data-timeout setting (default it to 3000)
* add a secondary button that sets this 0
* use that secondary button to test the timeout behaviour

Reply:
What is the advantage of using a button to test the timeout behaviour?

Reply:
Because we set the data- fields on buttons. So the secondary button would have a 0 timeout.